sql/pgwire: TestPipelineMetric failed

sql/pgwire.TestPipelineMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15288311?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15288311?buildTab=artifacts#/) on master @ [992d4573fb8b71e28717b4d14c4c7ebccb38c412](https://github.com/cockroachdb/cockroach/commits/992d4573fb8b71e28717b4d14c4c7ebccb38c412):


```
=== RUN   TestPipelineMetric
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/76e528b8ccf2c8b32dc6cf78a9848d00/logTestPipelineMetric2230920562
    test_log_scope.go:81: use -show-logs to present logs inline
I240516 04:24:48.939905 1 (gostd) rand.go:250  [-] 1  random seed: 693182867701299320
    conn_test.go:335: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:335
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestPipelineMetric
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/76e528b8ccf2c8b32dc6cf78a9848d00/logTestPipelineMetric2230920562
--- FAIL: TestPipelineMetric (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPipelineMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38801

sql/pgwire.TestPipelineMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15288774?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15288774?buildTab=artifacts#/) on master @ [992d4573fb8b71e28717b4d14c4c7ebccb38c412](https://github.com/cockroachdb/cockroach/commits/992d4573fb8b71e28717b4d14c4c7ebccb38c412):


```
=== RUN   TestPipelineMetric
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/76e528b8ccf2c8b32dc6cf78a9848d00/logTestPipelineMetric3942039491
    test_log_scope.go:81: use -show-logs to present logs inline
    conn_test.go:335: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:335
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestPipelineMetric
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/76e528b8ccf2c8b32dc6cf78a9848d00/logTestPipelineMetric3942039491
--- FAIL: TestPipelineMetric (0.37s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPipelineMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/pgwire.TestPipelineMetric [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15288790?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15288790?buildTab=artifacts#/) on master @ [992d4573fb8b71e28717b4d14c4c7ebccb38c412](https://github.com/cockroachdb/cockroach/commits/992d4573fb8b71e28717b4d14c4c7ebccb38c412):


```
=== RUN   TestPipelineMetric
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/76e528b8ccf2c8b32dc6cf78a9848d00/logTestPipelineMetric3026810578
    test_log_scope.go:81: use -show-logs to present logs inline
    conn_test.go:335: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:335
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestPipelineMetric
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/76e528b8ccf2c8b32dc6cf78a9848d00/logTestPipelineMetric3026810578
--- FAIL: TestPipelineMetric (0.02s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPipelineMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/pgwire.TestPipelineMetric [failed](https://tanzanite.cluster.engflow.com/invocations/default/e4f26551-16dd-4a97-a4b8-b14582ab9200?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsL3Bnd2lyZTpwZ3dpcmVfdGVzdA==) on master @ [021c72ae4a131bf511bff654fd3e4162947d3b15](https://github.com/cockroachdb/cockroach/commits/021c72ae4a131bf511bff654fd3e4162947d3b15):


```
=== RUN   TestPipelineMetric
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestPipelineMetric89487174
    test_log_scope.go:81: use -show-logs to present logs inline
    conn_test.go:335: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:335
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestPipelineMetric
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestPipelineMetric89487174
--- FAIL: TestPipelineMetric (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPipelineMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql/pgwire.TestPipelineMetric [failed](https://tanzanite.cluster.engflow.com/invocations/default/e993f35f-a005-4d1f-a3da-ff3164eedf16?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc3FsL3Bnd2lyZTpwZ3dpcmVfdGVzdA==) on master @ [021c72ae4a131bf511bff654fd3e4162947d3b15](https://github.com/cockroachdb/cockroach/commits/021c72ae4a131bf511bff654fd3e4162947d3b15):


```
=== RUN   TestPipelineMetric
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestPipelineMetric522237878
    test_log_scope.go:81: use -show-logs to present logs inline
    conn_test.go:335: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn_test.go:335
        	Error:      	Received unexpected error:
        	            	context canceled
        	Test:       	TestPipelineMetric
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestPipelineMetric522237878
--- FAIL: TestPipelineMetric (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPipelineMetric.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/pgwire/conn_test.go
+++ b/pkg/sql/pgwire/conn_test.go
@@ -244,7 +244,7 @@ func TestPipelineMetric(t *testing.T) {
 
 	// Run the conn's loop in the background - it will push commands to the
 	// buffer. serveImpl also performs the server handshake.
-	serveCtx, stopServe := context.WithCancel(ctx)
+	serveCtx, stopServe := context.WithCancel(context.Background())
 	defer stopServe()
 	serveGroup := ctxgroup.WithContext(serveCtx)
 	serveGroup.GoCtx(func(ctx context.Context) error {

